kpet/data.py

```diff
@@ -451,6 +474,56 @@ def convert_tree_arches(self):
 
             self.trees[name]["arches"] = list(tree_arches)
 
+    def convert_sets(self):
+        """ Convert each suite / cases sets from a list of regexes
+        to a list of set names matching those regexes
+
+        Returns:
+            Raises a schema.Invalid exception when finding an invalid regex
+        """
+
+        unassigned = {'unassigned'}
+        error = ("One of '{0}: set:' regexes\n" +
+                 "doesn't match any of the available sets\n" +
+                 "The regex: '{1}'\n" +
+                 "The avaliable sets: " + str(list(self.sets.keys())))
+
+        # TODO: Remove after conversion
+        # This function is to convert from the old format (where sets were a
+        # part of the pattern)
+        def pattern_conversion(pattern):
+            matches = set()
+            for set_name in self.sets:
+                target = Target(trees=Target.ANY,
+                                arches=Target.ANY,
+                                components=Target.ANY,
+                                sets=set(set_name),
+                                sources=Target.ANY)
+                if pattern.matches(target):
+                    matches.add(set_name)
+            return matches
+
+        for suite in self.suites:
+            suite_sets = set()
+            for regex in suite.sets:
+                match = set(filter(regex.fullmatch, self.sets.keys()))
+                if match == set():
+                    raise Invalid(error, "suite", regex.pattern)
+                suite_sets |= match
+            suite_sets |= pattern_conversion(suite.pattern)
+            for case in suite.cases:
+                case_sets = suite_sets.copy()
```

Comment:
Let's not do this. Let cases keep their own sets, don't copy the suite's sets, and update `Suite.matches()` to behave like `Case.matches()` does. This way we'll be able to reason about their applicability independently.

Reply:
Would't this mean the case sets weren't the union of their sets and the sets of their suite?
I thought we were aiming for that

Reply:
We take care of that in run.py: https://github.com/CKI-project/kpet/blob/f6749e35032eca50887e2229ef993fdcb354ede7/kpet/run.py#L127-L134

Reply:
Removed the copy and changed Suite.matches(), unsure about the Suite.matches change tho (if the new target is really needed there as I was getting the same results when testing it a few times, might just be due to the issue with an empty pattern tho)

Reply:
Also what did you mean byy making Suite.matches() behave like Case.matches because there are suites that don't have any sets specified like for example the kdump suite, wouldn't that mean the cases would never even be considered if the suite didn't have all the case sets?
I mean you mentioned that the case sets should be a subset of the suites sets, does that mean if he suite has no sets defined we assume all the sets or smth?
Yeah I think this is really confusing and should we need to agree on something and make the change either in kpet or in the database itself

Reply:
Sorry, this might be really confusing indeed. Could I talk to you 1-1 tomorrow? Will you be in the office? Right now I feel like I have porridge for my brain due to the cold :grin: